### DISCUSSION TITLE
Is it recommended to add .nicegui/storage_* to gitignore?

### DISCUSSION THREAD

**Discussion by s2terminal** [zauberzeug/nicegui#1658] [Q&A] [answered]
### Question

Hello. Thank you for the great library.

I have a question. I am writing a program based on the [authentications example](https://github.com/zauberzeug/nicegui/blob/main/examples/authentication/main.py). A file `.nicegui/storage_user_XXXXXXXX.json` is automatically generated. I don't think this is something that should be included in the development repository.

Is it recommended to add `.nicegui/storage_*` to gitignore?
Or is there an option to store user_storage elsewhere?

**Comment by falkoschindler** [comment, marked as answer]
Yes, I would recommend to ignore the .nicegui directory.
Alternatively you can set the [NICEGUI_STORAGE_PATH environment variable](https://nicegui.io/documentation#environment_variables) to something else.

**Comment by s2terminal** [comment]
Thank you. I did not know about setting environment variables.

I develop with Docker, so it looks like I should set it like `NICEGUI_STORAGE_PATH=/home/.nicegui` .

**Reply by rodja** [reply]
NiceGUI persists it's data into the `NICEGUI_STORAGE_PATH`. If you use docker, I recommend to make the path a volume so you do not loose the data if the container is rebuild.
